internal/devpkg/flakeref.go

```diff
@@ -260,6 +359,33 @@ func isArchive(path string) bool {
 		strings.HasSuffix(path, ".zip")
 }
 
+// buildOpaquePath escapes and joins path elements for a URL flakeref. The
+// resulting path is cleaned according to url.JoinPath.
+func buildOpaquePath(elem ...string) string {
```

Comment:
The Opaque word confused me. How about `buildEscapedPath` or `buildURLSafePath`?

Reply:
Sounds good. I'm going to rename it in another PR just so I don't have to rebase/merge all the other PRs.